manual/src/refman/patterns.etex

```diff
@@ -229,6 +344,13 @@ A pattern match must contain at least one value case. It is an error if
 all cases are exceptions, because there would be no code to handle
 the return of a value.
 
+\begin{caml_example}{toplevel}
+let range l =
+  match (List.hd l, List.hd (List.rev l)) with
+  | (f, l) -> Some (l - f)
+  | exception (Failure _) -> None;;
+\end{caml_example}
```

Comment:
I would complain if someone wrote this code without pattern-matching on the empty list first. Maybe:

```ocaml
let find_opt p li =
  match List.find p li with
  | exception Not_found -> None
  | v -> Some v
```

(The general theme being of calling a function that is known to return exceptions in some "standard" failure scenarios that cannot be easily discarded by cheap input verification. Yes, in many such cases we nowadays provide the `_opt` version itself in the standard library.)